File: objective-c/deploygate-ios-sdk-sample/SendLogViewController.m
//
//  SendLogViewController.m
//  deploygate-ios-sdk-sample
//
//  Created by kenta.imai on 2015/08/26.
//  Copyright (c) 2015年 DeployGate. All rights reserved.
//

#import "SendLogViewController.h"
#import <DeployGateSDK/DeployGateSDK.h>

@interface SendLogViewController ()
@property (strong, nonatomic) IBOutlet UITextView *textView;

@end

@implementation SendLogViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)sendLog:(UIBarButtonItem *)sender {
    NSString *text = self.textView.text;
    self.textView.text = @"";
    DGSLog(@"%@", text);

    [[[UIAlertView alloc] initWithTitle:@"DGSLog Done"
                                message:@"Require relaunch app"
                               delegate:self
                      cancelButtonTitle:@"Continue"
                      otherButtonTitles:@"Relaunch", nil] show];
}

- (void)alertView:(UIAlertView *)alertView clickedButtonAtIndex:(NSInteger)buttonIndex
{
    if (buttonIndex == 1) {
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wundeclared-selector"
        [[UIApplication sharedApplication] performSelector:@selector(terminateWithSuccess)];
#pragma clang diagnostic pops
    }
}

@end

File: swift/deploygate-ios-sdk-swift-sample/SendLogViewController.swift
//
//  SendLogViewController.swift
//  deploygate-ios-sdk-swift-sample
//
//  Created by kenta.imai on 2015/08/31.
//  Copyright (c) 2015年 DeployGate. All rights reserved.
//

import UIKit

class SendLogViewController: UIViewController {
    @IBOutlet var textView: UITextView!

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    @IBAction func sendLog(sender: AnyObject) {
        let text = textView.text
        textView.text = ""

        DGSLogv("%@", getVaList([text]))
        let alert:UIAlertController = UIAlertController(title: "DGSLog Done", message: "Require relaunch app", preferredStyle: UIAlertControllerStyle.Alert)
        alert.addAction(UIAlertAction(title: "Continue", style: UIAlertActionStyle.Cancel, handler:nil))
        alert.addAction(UIAlertAction(title: "Relaunch",
            style: UIAlertActionStyle.Default,
            handler:{
                (action:UIAlertAction) -> Void in
                exit(0) // finish application
        }))
        presentViewController(alert, animated: true, completion: nil)
    }
}
